src/marqo/core/models/marqo_index.py

```diff
@@ -123,6 +123,24 @@ class Model(StrictBaseModel):
     properties: Optional[Dict[str, Any]]
     custom: bool = False
 
+    @root_validator(pre=False)
+    def validate_custom_properties(cls, values):
+        """Validate custom model properties.
+
+        Raises:
+            InvalidArgumentError: If model properties are invalid.
+        """
+        model_name = values.get('name')
+        properties = values.get('properties')
+        custom = values.get('custom')
+        if properties and custom:
+            try:
+                s2_inference.validate_model_properties(model_name, properties)
+            except InvalidModelPropertiesError as e:
+                raise InvalidArgumentError(
```

Comment:
The convention for pydantic validators is to raise ValueError. API will handle this\